public/app/features/explore/ShortLinkButtonMenu.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react';
+
+import { IconName, urlUtil } from '@grafana/data';
+import { reportInteraction } from '@grafana/runtime';
+import { ToolbarButton, Dropdown, Menu, Stack, ToolbarButtonRow } from '@grafana/ui';
+import { t } from 'app/core/internationalization';
+import store from 'app/core/store';
+import { createAndCopyShortLink } from 'app/core/utils/shortLinks';
+import { useSelector } from 'app/types';
+
+import { getUrlStateFromPaneState } from './hooks/useStateSync';
+import { selectPanes } from './state/selectors';
+
+export function ShortLinkButtonMenu() {
+  const panes = useSelector(selectPanes);
+  const [isOpen, setIsOpen] = useState(false);
+  const onCopyShortLink = (persistType?: string, url?: string) => {
+    if (persistType !== undefined) {
+      store.set('grafana.explore.shortLinkShareType', persistType);
+    }
+    createAndCopyShortLink(url || global.location.href);
+    reportInteraction('grafana_explore_shortened_link_clicked');
+  };
+
+  interface ShortLinkMenuItemData {
+    key: string;
+    label: string;
+    icon: IconName;
+    getUrl: Function;
+    description?: string;
+  }
+
+  const menuOptions: ShortLinkMenuItemData[] = [
+    {
+      key: 'copy-link',
+      label: t('explore.toolbar.copy-shortened-link', 'Copy shortened link'),
+      icon: 'link',
+      getUrl: () => undefined,
+    },
+    {
+      key: 'copy-link-abs-time',
+      label: t('explore.toolbar.copy-shortened-link-abs-time', 'Copy shortened link with absolute time'),
+      icon: 'clock-nine',
+      description: t(
+        'explore.toolbar.copy-shortened-link-abs-time-desc',
+        'Locks the current time for consistent viewing'
```

Comment:
> Locks...

Could just be me, but I find it confusing what "Locks..." actually mean here. At first sight I thought it means it may change my time range in the Explore.

Reply:
@diegoadams I didnt have it change the selected time range in explore because it seemed like an unrelated action. Should we choose different verbiage here?

Reply:
> @diegoadams I didnt have it change the selected time range in explore because it seemed like an unrelated action. Should we choose different verbiage here?

Perhaps we could try :

- Preserve Time Range 
- Freeze Current Time
- Fix Time to Current Selectio

Reply:
How about "Changes relative time to absolute in the URL"? or "Uses absolute time in the URL"?